syntax = "proto3";


service SchedulerService {
    rpc JobState(JobStateMessage) returns (ReplyResult) {

    }

    rpc Predictor(JobCombine) returns (ReplyPrediction) {

    }

    rpc Load(LoadInformation) returns (ReplyResult) {

    }

    rpc Regist(NodeInformation) returns (ReplyResult){

    }
}

service WorkerService {
    rpc AccpetJob(JobInformation) returns (ReplyResult) {

    }
}

message JobStateMessage {
    string type = 1;
    int32 JobID = 2;
}

message ReplyResult {
    string response = 1;
}

message JobCombine {
    int32 partition = 1;
    repeated int32 JobIDs =2;
    repeated int32 MPSPercentage = 3;
}

message ReplyPrediction {
    repeated int32 JobIDs =1 ;
    repeated float Prediction = 2;
}

message LoadInformation {
    int32 GPU_ID = 1;
    int32 GI_ID = 2;
    float load = 3;
    string name = 4;
}

message JobInformation {
    int32 GPU_ID = 1;
    int32 JobID = 2;
}


message NodeInformation {
    string name = 1;
    string ip = 2;
    int32 port = 3;
}